specifyweb/specify/views.py

```diff
@@ -399,86 +400,156 @@ class ReplaceRecordPT(PermissionTarget):
     update = PermissionTargetAction()
     delete = PermissionTargetAction()
 
+@transaction.atomic
+def record_merge_fx(model_name: str, old_model_id: int, new_model_id: int) -> http.HttpResponse:
+    """Replaces all the foreign keys referencing the old record ID
+    with the new record ID, and deletes the old record record.
+    """
+    # Confirm the target model table exists
+    target_model = apps.get_model('specify', model_name.lower())
+    if target_model is None:
+        return http.HttpResponseNotFound("model_name: " + model_name + "does not exist.")
+
+    # Check to make sure both the old and new agent IDs exist in the table
+    if not target_model.objects.filter(id=old_model_id).select_for_update().exists():
+        return http.HttpResponseNotFound(model_name + "ID: " + str(old_model_id) + " does not exist.")
+    if not target_model.objects.filter(id=new_model_id).select_for_update().exists():
+        return http.HttpResponseNotFound(model_name + "ID: " + str(new_model_id) + " does not exist.")
+    
+    # Get dependent fields and objects of the target object
+    target_object = target_model.objects.get(id=old_model_id)
+    dependant_table_names = [rel.relatedModelName
+        for rel in target_object.specify_model.relationships
+        if api.is_dependent_field(target_object, rel.name)]
+
+    # Get all of the columns in all of the tables of specify the are foreign keys referencing AgentID
+    foreign_key_cols = []
+    for table in spmodels.datamodel.tables:
+        for relationship in table.relationships:
+            if relationship.relatedModelName.lower() == model_name.lower():
+                foreign_key_cols.append((table.name, relationship.name))
```

Comment:
Is this the best way of doing this? I am not found of a for-loop in a for-loop (that's a lot of iterations!),  but I can not immediately think of a better solution, and if it works--- it works.  
Perhaps this should be looked into further, but I will put some more thought into it nonetheless.

Reply:
ya, this could maybe have been written in a functional manner, but it would be doing essentially the same thing.